docs/source/api/notebooks/Notebook_5_GP_solution.ipynb

```diff
@@ -0,0 +1,631 @@
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Tip:</b> We also provide a Python script to generate synthetic supernode-based graph data for user to better understand the `super-node` grahp data format, which is avaible <a href=\"https://github.com/awslabs/graphstorm/blob/main/docs/source/api/notebooks/dummy_gc_data.py\"><u>here</u></a>.</div>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can download the source code of the OGBG conversion and generate the `super-node` format OGBG data with the following commands.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!wget -c https://raw.githubusercontent.com/awslabs/graphstorm/main/examples/graph_prediction/gen_ogbg_supernode.py\n",
+    "\n",
+    "!python gen_ogbg_supernode.py --ogbg-data-name molhiv --output-path ./supernode_raw/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The converted OGBG data will be stored at `./supernode_raw/`. Then we can run GraphStorm's gconstruct command to partition the graph for model training and inference. The processed graph is stored in the `/supernode_gs_1p/` folder."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!python -m graphstorm.gconstruct.construct_graph \\\n",
+    "        --conf-file ./supernode_raw/config.json \\\n",
+    "        --output-dir ./supernode_gs_1p/ \\\n",
+    "        --num-parts 1 \\\n",
+    "        --graph-name supernode_molhiv"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-warning\">\n",
+    "<b>Tip:</b> If the above Python commands fail to run in a Notebook, please run it in a Linux terminal environment where GraphStorm is install.</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Customized GNN Encoder for Graph Prediction\n",
+    "\n",
+    "The key component of this `super-node` based solution is the GNN model that can perform message passing and aggregation in each subgraph, and then perform sorts of `Read-out` operation in the super nodes. This component could be easily implemented as a customized GraphStorm GNN Encoder like demonstrated in the [Notebook 4: Customized Models](https://graphstorm.readthedocs.io/en/latest/api/notebooks/Notebook_4_Customized_Models.html)\n",
+    "\n",
+    "As shown in the diagram below, a **super node** will aggregate the representations from other nodes in each GNN layer. Built-in GNN encoders will udpate the aggregated representations with an additional trainable parameters. This operation is different from the common `Read-out` operation, hence causing worse performance.\n",
```

Comment:
Worse performance in terms of accuracy or runtime/computational cost? Let's try to be specific

Reply:
Updated to `hence potentially causing worse model performance in graph-level prediction.`